### PR TITLE
Update the email comparison page to use a stacked card layout and embed account creation

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* This PR implements the basic design for the new Email Comparison page from pcDL0W-79-p2, which includes a stacked card layout and account creation from within the page.
Note that this PR depends on #51217 for the Email account creation component in that PR, and existing logic for Google Workspace. As such, some of the form fields from that PR may change slightly. (The most likely change is to increase the height to support the password reset field.)

You can access the new layout by navigating to the email comparison page for a domain without email (`/email/:domain/manage/:siteSlug?titan/provision-mailboxes` or via Upgrades -> Domains -> Add (next to a domain), and then adding `?flags=titan/provision-mailboxes` to the URL.) **Note:** _this flag has been updated from an earlier version of this PR_

The following implementation items are not covered by this PR:
 * Implement Email Forwarding creation logic -- we may take this on in a later PR, but it's not critical
 * Implement state to indicate email can't be purchased for the domain
 * Implement states where one or more of the options can't be purchased
 * Implement handling for cases where the domain isn't eligible for the free trial -- we'll take this on later this week, as we don't need this to launch this functionality to users as-is

#### Screenshots

**Before:**

Existing email comparison page

**After:**

_Initial, unexpanded state_
Initial state with Email expanded

_Google Workspace expanded_
Google Workspace expanded

_Email Forwarding expanded_
Email Forwarding expanded

#### Testing instructions

Verify that you can add all three types of email account via this UI.

### CODE REVIEW

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 1]
```diff
@@ -296,20 +272,290 @@ class EmailProvidersComparison extends React.Component {
...
+	getTitanFeatures() {
+		const { translate } = this.props;
+
+		return [
+			translate( 'Monthly billing' ),
+			translate( 'Send and receive from your custom domain' ),
+			translate( '30GB storage' ),
+			translate( 'Email, calendars, and contacts' ),
+			translate( 'One-click import of existing emails and contacts' ),
+			translate( 'Apps for Android and iOS' ),
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **4** times:
`translate( 'Jetpack for Android and iOS' )` *ES Score: 7*

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 2]
```diff
@@ -296,20 +272,290 @@ class EmailProvidersComparison extends React.Component {
...
+		}
+
+		const googleProps = this.getGoogleDetailProps();
+		// If we don't have any users, initialize the list to have 1 empty user
+		const googleUsers =
+			( this.state.googleUsers ?? [] ).length === 0
+				? newUsers( domain.name )
+				: this.state.googleUsers;
+
+		const formattedPrice = translate( '{{price/}} /user /month billed yearly', {
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( '%(price)s per month, billed yearly' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/updated-email-comparison-page)

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 3]
```diff
@@ -244,68 +248,294 @@ class EmailProvidersComparison extends React.Component {
...
-			? googleWorkspaceIcon
-			: gSuiteLogo;
+		const priceArguments = {
+			components: {
+				price: <span>{ getMonthlyPrice( gSuiteProduct?.cost ?? null, currencyCode ) }</span>,
+			},
+			comment: '{{price/}} is the formatted price, e.g. $20',
+		};
+		const formattedPrice = config.isEnabled( 'email/centralized-home' )
+			? translate( '{{price/}} /user /month billed annually', priceArguments )
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx:301]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'per month{{br/}}billed yearly' )` *ES Score: 6*

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 4]
```diff
@@ -244,68 +263,319 @@ class EmailProvidersComparison extends React.Component {
...
 	}
 
+	renderStackedForwardingDetails() {
+		const { domain, translate } = this.props;
+
+		const formFields = (
+			<>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Emails sent to' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Emails sent to', { context: 'Form label: Email sent to [address] (will arrive at...)'} )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/updated-email-comparison-page)

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 5]
```diff
@@ -244,68 +263,319 @@ class EmailProvidersComparison extends React.Component {
...
+						<FormTextInputWithAffixes
+							required
+							suffix={ `@${ domain.name }` }
+							onChange={ this.onForwardingEmailChange }
+						/>
+					</FormLabel>
+				</FormFieldset>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Will be forwarded to' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Will be forwarded to', { context: 'Form label: (Email sent to [address]) will arrive at [address]'} )` *ES Score: 15*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/updated-email-comparison-page)

File: client/lib/titan/new-mailbox.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+	}
+
+	return { value, error };
+};
+
+const validateMailboxName = ( { value, error } ) => ( {
+	value,
+	error:
+		! error && ! /^[0-9a-z_-](\.?[0-9a-z_-])*$/i.test( value )
+			? translate( 'Only numbers, letters, dashes, underscores, and periods are allowed.' )
```
Comment (a8ci18n) [inline client/lib/titan/new-mailbox.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Only numbers, letters, dashes, underscores, apostrophes and periods are allowed.' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/updated-email-comparison-page)

**Review by hambai — COMMENTED** [review]
Hmm - I tried testing this but the `areAllMailboxesValid` returns false (I guess because `alternativeEmail` is empty). Is that intentional?

**Comment by daledupreez** [comment]
> Hmm - I tried testing this but the `areAllMailboxesValid` returns false (I guess because `alternativeEmail` is empty). Is that intentional?

Definitely not intentional -- the validation should allow `alternativeEmail` to be empty if we're not showing the field (which is the case while we're adding it on the back end.

I'll do some testing and exploring in the morning to see what's going on.

**Comment by daledupreez** [comment]
I am closing this _massive_ PR in favour of #51298 and #51307.

**Pull request closed without merging**